### PR TITLE
Delete unused files of thirdparty libs (zlib, mbedtls)

### PR DESCRIPTION
**Pull request by ghost**
They weren't listed inside appropriate source lists (inside `SCsub` files) so this change shouldn't affect compiled binaries at all but obviously check it

### CODE REVIEW

**Comment by fire** [comment]
Please update the readme with the deletion indtructions

File: thirdparty/README.md [hunk 1]
```diff
@@ -556,8 +556,11 @@ File extracted from upstream release tarball:
 
 - All `.h` from `include/mbedtls/` to `thirdparty/mbedtls/include/mbedtls/`
   and all `.h` from `include/psa/` to `thirdparty/mbedtls/include/psa/`
-- All `.c` and `.h` from `library/` to `thirdparty/mbedtls/library/` except
-  for the `psa_*.c` source files
+- All `.c` and `.h` from `library/` to `thirdparty/mbedtls/library/`
+- From `library/` to `thirdparty/mbedtls/library/`:
+  - All `.c` and `.h` files
+  - Except `psa_*.c`, `bignum_mod.c`, `block_cipher.c`, `ecp_curves_new.c`,
```
Comment (akien-mga) [inline thirdparty/README.md:562]: Actually we're now using the `psa_*.c` files after #96394, @Faless just omitted to update the instructions. Could you remove the exception for `psa_*.c` in this PR too?

**Review by akien-mga — APPROVED** [review]

**Review by fire — APPROVED** [review]
The GitHub actions integration test still functions, the files are deleted, and the readme instructions are updated. It looks good to me.

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**